Redraw commutative diagrams with Tikz

platipo opened the pull request:
Fix #214

platipo commented:
@BartoszMilewski can you do a little review and give you an opinion?

BartoszMilewski commented:
I don't have a good way of viewing these changes

platipo commented:
Maybe the book is more useful 

[category-theory-for-programmers.pdf](https://github.com/hmemcpy/milewski-ctfp-pdf/files/3694298/category-theory-for-programmers.pdf)

BartoszMilewski commented:
I had a look at some of the proposed changes and overall I prefer the originals. Sorry!

platipo commented:
For simplicity I've added all figures in [overleaf](https://www.overleaf.com/read/nngzhdpbzhfb) and modified them locally.

@BartoszMilewski do you prefer all originals or just some of them?

BartoszMilewski commented:
I can see overleaf, but it would be easier if I could see them side-by-side. Could you embed originals? Also, could you split the file--the recompilation tends to time out on large files.

platipo commented:
Unfortunately compiling side-by-side would require a gigantic amount of work: uploading the images, creating a side-by-side macro and getting it to work, split various file in various parts...

I'm aware of the slowness caused by compilation and in the future I'm planning to split the thing in various files. For the moment I'm sticking with it because this is the fastest method.

platipo commented:
> I had a look at some of the proposed changes and overall I prefer the originals. Sorry!

Which are the ones you don't like?

platipo commented:
> In naturality: Functor F acting on object or morphisms should not use the subscript. It's F a, not F_a, and so on.

So I was writing all formulas wrong, is it `$Fa$` in LaTeX?

BartoszMilewski commented:
> Unfortunately compiling side-by-side would require a gigantic amount of work: uploading the images, creating a side-by-side macro and getting it to work, split various file in various parts...
I meant in Overleaf. It's easy to upload and embed pictures. The hard part is finding them, and that's what I'm struggling with.

BartoszMilewski commented:
> > In naturality: Functor F acting on object or morphisms should not use the subscript. It's F a, not F_a, and so on.
> 
> So I was writing all formulas wrong, is it `$Fa$` in LaTeX?

I write it as `$F a$` (with a space)

platipo commented:
> > Unfortunately compiling side-by-side would require a gigantic amount of work: uploading the images, creating a side-by-side macro and getting it to work, split various file in various parts...
> 
> I meant in Overleaf. It's easy to upload and embed pictures. The hard part is finding them, and that's what I'm struggling with.

For the moment I added a comment with the image name and I've crated a local index to browse them using symbolic links.

platipo commented:
[category-theory-for-programmers-g497eda4.pdf](https://github.com/hmemcpy/milewski-ctfp-pdf/files/3696265/category-theory-for-programmers-print.pdf)

Page modified are:
 - 109
 - 116
 - 117
 - 119
 - 129
 - 178
 - 179
 - 223
 - 224
 - 224
 - 227
 - 228
 - 229
 - 230
 - 247
 - 248
 - 249
 - 250
 - 252
 - 322
 - 323
 - 324
 - 325
 - 326
 - 327

drupol commented:
If you rebase your branch and thanks to #251, you will have the PDFs automatically generated, maybe it would be nice to do it so we can see the result directly in the PDFs ?
